microsoft-edge/devtools-guide-chromium/whats-new/2023/01/devtools-109.md

```diff
@@ -0,0 +1,201 @@
+
+
+<!-- ====================================================================== -->
+## Import and export your DevTools instances with the new enhanced traces feature
+
+<!-- Subtitle: Use enhanced traces to preserve Console messages, DOM snapshots, and the JavaScript running on the page when exporting a heap snapshot or performance trace. -->
+
+In Microsoft Edge 109, you can turn on new settings to export enhanced traces from the **Performance** and **Memory** tools.  These enhanced traces include more information, such as:
+*  Messages logged to the Console.
+*  The JavaScript that was running on the page at the time of recording.
+*  Snapshots of the DOM.
+
+When saving performance profiles, heap snapshots, allocation timelines, or allocation sampling, you can now export a new `.devtools` file format.  When you then import the `.devtools` file, a new instance of DevTools opens, with the state of your **Console**, **Sources**, and **Elements** tools preserved.  These enhanced traces offer a powerful new way to collaborate and share the information in your DevTools.
+
+To try the enhanced traces feature:
+
+1. In DevTools, click the **Settings** (![Settings icon](../../../media/settings-gear-icon-light-theme.png)) button.
+
+1. In the **Persistence** section of the **Preferences** settings page, select the **Export enhanced performance and memory traces** checkbox:
+
+   ![Settings > Preferences > Persistence checkboxes](./devtools-109-images/enhanced-trace-settings.png)
+
+1. If you want to preserve Console messages, JavaScript, or DOM snapshots, select the corresponding checkboxes.
+
+1. Click the **Close** (**x**) button in DevTools **Settings**.
+
+1. In the **Performance** tool, take a recording.  Or, in the **Memory** tool, take a heap snapshot.
+
+1. If you use the **Performance** tool, right-click the **Save profile** (![Save profile icon](./devtools-109-images/save-profile-icon.png)) button, and then select **.devtools (enhanced format for Microsoft Edge)**:
+
+   ![Exporting a trace](./devtools-109-images/enhanced-trace-export.png)
+
+1. In the **Save As** dialog, save the new `.devtools` file.
+
+   Or, in the **Memory** tool, click the **Save** link next to the heap snapshot.  In the **Save As** dialog, save the new `.devtools` file.
+
+**Importing:**
+
+1. Import the `.devtools` file from within the **Performance** tool or **Memory** tool.  If you use the **Performance** tool, click the the **Load profile** (![Load profile icon](./devtools-109-images/load-profile-icon.png)) button:
+
+   ![Importing a trace from within the Performance tool](./devtools-109-images/enhanced-trace-import.png)
+
+   If you use the **Memory** tool, right-click the **Heap Snapshots** pane and then select **Load**.
+
+1. A new DevTools window opens, containing a subset of the tools, including the **Performance** or **Memory** tool.  The **Elements**, **Console**, and **Sources** tools are pre-populated with the preserved state:
```

Comment:
Is "or" correct?  
`including the **Performance** or **Memory** tool"`

Reply:
No I think it's "and". Both tools appear in the new window.

Reply:
It's both but in this example, only the Performance tool is populated.

```suggestion
1. A new DevTools window opens, containing a subset of the tools, including the **Performance** tool.  The **Elements**, **Console**, and **Sources** tools are pre-populated with the preserved state:
```

Reply:
trying and/or, eg:

A new DevTools window opens, containing a subset of the tools, including the **Performance** and/or **Memory** tool.